### ISSUE TITLE
"Failed to load local shard" on service startup due to ./storage/collections/.DS_Store

### ISSUE THREAD

**Issue by tot-ra** [qdrant/qdrant#4939]
If user on Mac opened qdrant folder and got `.DS_Store` automatically generated, then qdrant no longer starts

## Current Behavior
```
qdrant-1  |            _                 _
qdrant-1  |   __ _  __| |_ __ __ _ _ __ | |_
qdrant-1  |  / _` |/ _` | '__/ _` | '_ \| __|
qdrant-1  | | (_| | (_| | | | (_| | | | | |_
qdrant-1  |  \__, |\__,_|_|  \__,_|_| |_|\__|
qdrant-1  |     |_|
qdrant-1  |
qdrant-1  | Version: 1.11.0, build: 63363956
qdrant-1  | Access web UI at http://localhost:6333/dashboard
qdrant-1  |
qdrant-1  | 2024-08-22T16:26:53.568247Z  INFO storage::content_manager::consensus::persistent: Loading raft state from ./storage/raft_state.json
qdrant-1  | 2024-08-22T16:26:53.584887Z  WARN storage::content_manager::toc: Collection config is not found in the collection directory: ./storage/collections/.DS_Store, skipping
qdrant-1  | 2024-08-22T16:26:53.584995Z  INFO storage::content_manager::toc: Loading collection: collection_768
qdrant-1  | 2024-08-22T16:26:53.653409Z ERROR qdrant::startup: Panic backtrace:
qdrant-1  |    0: std::backtrace::Backtrace::create
qdrant-1  |    1: qdrant::startup::setup_panic_hook::{{closure}}
qdrant-1  |    2: std::panicking::rust_panic_with_hook
qdrant-1  |    3: std::panicking::begin_panic_handler::{{closure}}
qdrant-1  |    4: std::sys_common::backtrace::__rust_end_short_backtrace
qdrant-1  |    5: rust_begin_unwind
qdrant-1  |    6: core::panicking::panic_fmt
qdrant-1  |    7: collection::shards::shard_holder::ShardHolder::load_shards::{{closure}}
qdrant-1  |    8: collection::collection::Collection::load::{{closure}}
qdrant-1  |    9: storage::content_manager::toc::TableOfContent::new
qdrant-1  |   10: qdrant::main
qdrant-1  |   11: std::sys_common::backtrace::__rust_begin_short_backtrace
qdrant-1  |   12: main
qdrant-1  |   13: <unknown>
qdrant-1  |   14: __libc_start_main
qdrant-1  |   15: _start
qdrant-1  |
qdrant-1  | 2024-08-22T16:26:53.653455Z ERROR qdrant::startup: Panic occurred in file /qdrant/lib/collection/src/shards/replica_set/mod.rs at line 277: Failed to load local shard "./storage/collections/collection_768/0": Service internal error: Not a directory (os error 20)
```

**Comment by timvisee** [comment]
Nice find. Thanks for the report!

I think we'll have to ignore non-directories.

We can also ignore everything prefixed with a dot. Though I'm afraid that'd break collections that have such name on purpose.

**Comment by jayantpranjal0** [comment]
Heyy, @tot-ra I tried to reproduce this issue, on my system, and this didn't happen on my system . Can you list down the steps to reproduce this?

**Comment by timvisee** [comment]
> Heyy, @tot-ra I tried to reproduce this issue, on my system, and this didn't happen on my system . Can you list down the steps to reproduce this?

Did you have a `.DS_Store` file in `storage/collections/`?

You can create it by opening that directory in Finder on macOS and then drag a directory around. Otherwise you can probably do `touch storage/collections/.DS_Store`.

**Comment by jayantpranjal0** [comment]
Yeahh, I made two files and moved it around to get a .DS_Store file in the folder.
![image](https://github.com/user-attachments/assets/f5e618f3-0b09-41c7-ae3f-575ca53f1484)

**Comment by jayantpranjal0** [comment]
@tot-ra can you list down how to reproduce this?
Also, @timvisee , PR were made for this issue.

**Comment by timvisee** [comment]
You're right. Thanks for diving into the details.

It could very well be a permission error, or some other kind of disk corruption. But if `.DS_Store` in the collections directory would cause issues we'd be seeing a different type of error.

Closing this now.

Please feel free to open it again if the problem persists.

**Comment by richarddd** [comment]
I have the same issue. Running this fixed it:
`find . -name '.DS_Store' -type f -delete`

This should be reopened

**Comment by timvisee** [comment]
We've just released [Qdrant 1.13.5](https://github.com/qdrant/qdrant/releases/tag/v1.13.5) which includes the fix for this issue.

Manually removing the `.DS_Store` files should now not be necessary anymore if running the newer version.
